### PR TITLE
Fixed tests

### PR DESCRIPTION
**Pull request by moomindani**
### Description

This PR fixes 12 test cases in the following test scripts: 
- `tests/functional/adapter/test_basic.py`
- `tests/functional_test/adapter/test_basic_functional_test.[y`
- `tests/single_functional/adapter/single_test.py`
- `tests/unit/ttest_glue_session.py`
- `tests/unit/test_glue_session_demo.py`

### Test

```
% python3 -m pytest -s 
============================================================================================================ test session starts =============================================================================================================
platform darwin -- Python 3.9.6, pytest-7.4.0, pluggy-1.3.0
rootdir: /Users/sekiyama/Documents/workspace/dbt-glue
plugins: csv-3.0.0, logbook-1.2.0, flaky-3.7.0, dotenv-0.5.2, xdist-3.3.1, typeguard-2.13.3
collected 12 items    
...
====================================================================================================== 12 passed in 1700.40s (0:28:20) =======================================================================================================

```

### Checklist

- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have updated the `CHANGELOG.md` and added information about my change to the "dbt-glue next" section.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tests/util.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+
+def get_s3_location():
+    if "DBT_S3_LOCATION" in os.environ:
+        return os.environ.get("DBT_S3_LOCATION")
+    else:
+        raise ValueError("DBT_S3_LOCATION must be configured")
+
+def get_role_arn():
+    return os.environ.get("DBT_GLUE_ROLE_ARN", f"arn:aws:iam::{get_account_id()}:role/GlueInteractiveSessionRole")
```
Comment (aajisaka) [inline tests/util.py:26]: Would you replace `DBT_GLUE_ROLE_ARN` with `DBT_GLUE_ARN` to match with readme?

File: tests/unit/conftest.py [hunk 2]
```diff
@@ -43,8 +42,8 @@ def clean():
 @pytest.fixture(scope="module")
 def credentials():
     return GlueCredentials(
-        role_arn=f"arn:aws:iam::{account}:role/GlueInteractiveSessionRole",
-        region=region,
+        role_arn=get_role_arn(),
+        region=get_region(),
```
Comment (aajisaka) [inline tests/unit/conftest.py:46]: Would you update the `session_provisioning_timeout_in_seconds` to 60 or 120 seconds? 30 second is a bit short and it sometimes timeout.

Comment (moomindani) [reply tests/unit/conftest.py:46]: fixed

File: dbt/adapters/glue/gluedbapi/connection.py [hunk 3]
```diff
@@ -45,14 +45,14 @@ def __init__(self, credentials: GlueCredentials, session_id_suffix: str = None,
 
     def _connect(self):
         logger.debug("GlueConnection connect called")
-        if not self.session_id:
```
Comment (aajisaka) [inline dbt/adapters/glue/gluedbapi/connection.py:48]: This change is not necessary. If `self.session_id` is `None`, `not self.session_id` returns `True`.

File: dbt/adapters/glue/gluedbapi/connection.py [hunk 4]
```diff
@@ -276,7 +276,7 @@ def cursor(self, as_dict=False) -> GlueCursor:
         if self.state == GlueSessionState.READY:
             self._init_session()
             return GlueDictCursor(connection=self) if as_dict else GlueCursor(connection=self)
-        else:
```
Comment (aajisaka) [inline dbt/adapters/glue/gluedbapi/connection.py:279]: This change is not necessary. `not self.session_id or self.session_id is not None` always returns `True`.

File: dbt/adapters/glue/gluedbapi/connection.py [hunk 5]
```diff
@@ -286,14 +286,15 @@ def cursor(self, as_dict=False) -> GlueCursor:
...
-                    logger.debug(f"session is already stopped or failed")
+                    logger.debug(f"session {self.session_id} is already stopped or failed")
             except Exception as e:
                 raise e
         
 
     def close_session(self):
         logger.debug("GlueConnection close_session called")
-        if not self._session:
+        if not self._session or not self.session_id or self.session_id is None:
```
Comment (aajisaka) [inline dbt/adapters/glue/gluedbapi/connection.py:296]: `or self.session_id is None` can be removed

File: dbt/adapters/glue/gluedbapi/connection.py [hunk 6]
```diff
@@ -315,16 +316,16 @@ def state(self):
         if self._state in [GlueSessionState.FAILED]:
             return self._state
         try:
-            if not self.session_id:
+            if not self.session_id or self.session_id is None:
```
Comment (aajisaka) [inline dbt/adapters/glue/gluedbapi/connection.py:319]: This change can be removed

**Pull request merged**